packages/forms-web-app/src/dynamic-forms/dynamic-components/multi-file-upload/question.js

```diff
@@ -10,6 +32,146 @@ class MultiFileUploadQuestion extends Question {
 			fieldName,
 			validators
 		});
+
+		if (documentType) {
+			this.documentType = documentType;
+		} else {
+			// default doc type
+			this.documentType = documentTypes.dynamic;
+		}
+	}
+
+	/**
+	 * Save an uploaded file
+	 * @param {ExpressRequest} req
+	 * @param {ExpressResponse} res
+	 * @param {Journey} journey
+	 * @param {Section} sectionObj
+	 * @param {JourneyResponse} journeyResponse
+	 * @returns
+	 */
+	saveAction = async (req, res, journey, sectionObj, journeyResponse) => {
+		try {
+			const { body } = req;
+			// const { errors = {}, errorSummary = [] } = body;
+			const encodedReferenceId = encodeURIComponent(journeyResponse.referenceId);
+
+			// set answer on response
+			let responseToSave = {
+				answers: {
+					[this.fieldName]: {
+						uploadedFiles: []
+					}
+				}
+			};
+
+			// Ensure we are always working with an array. Single files would otherwise be an object.
+			if (!Array.isArray(req.files[this.fieldName])) {
+				req.files[this.fieldName] = [req.files[this.fieldName]];
+			}
+
+			// remove files from response
+			if ('removedFiles' in body) {
+				const removedFiles = JSON.parse(body.removedFiles);
+
+				journeyResponse.answers[this.fieldName].uploadedFiles = removeFiles(
+					journeyResponse.answers[this.fieldName].uploadedFiles,
+					removedFiles
+				);
+			}
+
+			// save files to blob storage
+			const uploadedFiles = await this.#saveFilesToBlobStorage(req, journeyResponse);
+
+			// add saved docs to response
+			responseToSave.answers[this.fieldName].uploadedFiles = [];
+
+			if (journeyResponse?.answers[this.fieldName]?.uploadedFiles) {
+				responseToSave.answers[this.fieldName].uploadedFiles.push(
+					...journeyResponse.answers[this.fieldName].uploadedFiles
+				);
+			}
+
+			if (uploadedFiles) {
+				responseToSave.answers[this.fieldName].uploadedFiles.push(...uploadedFiles);
+			}
+
+			// save answer to database
+			await patchQuestionResponse(journeyResponse.journeyId, encodedReferenceId, responseToSave);
+
+			// todo: validation after saving required?
+			// if (Object.keys(errors).length > 0) {
+			// 	const answer = journeyResponse.answers[this.fieldName].uploadedFiles;
+			// 	let viewModel = journey.getQuestionViewModel(sectionObj, this, answer);
+			// 	viewModel.errors = errors;
+			// 	// multi-file upload validation would otherwise map these errors individual to e.g.
+			// 	// `#files.supporting-documents[3]` which does not meet the gov uk presentation requirements.
+			// 	viewModel.errorSummary = errorSummary.map((error) => ({
+			// 		...error,
+			// 		href: '#'
+			// 	}));
+			// 	return this.#showError(res, viewModel);
+			// }
+
+			// map the response back to journeyResponse
+			journeyResponse.answers[this.fieldName] = responseToSave.answers[this.fieldName];
+
+			// update journey based on response
+			const updatedJourney = new journey.constructor(journeyResponse);
```

Comment:
wasn't sure on the best way to achieve this, alternative could be to have an update journey function

Reply:
I think thats a good thing to keep in mind if we need to do this anywhere else in the future, but if this behaviour is just a one off then this should be fine.